### PR TITLE
refactor(RHTAPWATCH-479): kwok image use sysv-init

### PR DESCRIPTION
**Pull request by ifireball**
Make our kwok image use the busybox version of sys-v init to launch and monitor processes. This will allow us to make it run multiple processes.

### CODE REVIEW

File: kwok/Dockerfile [hunk 1]
```diff
@@ -4,11 +4,12 @@ ARG CLUSTER_NAME=kwok
...
+COPY --chmod=644 inittab /etc/inittab
 
 RUN KWOK_KUBE_APISERVER_PORT=0 kwokctl create cluster --name "$CLUSTER_NAME" || exit 1 && \
     kwokctl --name="$CLUSTER_NAME" kubectl apply -f toolchain.dev.openshift.com_usersignups.yaml && \
     sh ./user-signups.sh && \
     rm user-signups.sh toolchain.dev.openshift.com_usersignups.yaml && \
     kwokctl stop cluster --name="$CLUSTER_NAME"
 
-ENTRYPOINT ["/entrypoint.sh"]
+ENTRYPOINT ["/sbin/init"]
```
Comment (yftacherzog) [inline kwok/Dockerfile:15]: If I understand correctly, the `ENTRYPOINT ["/entrypoint.sh"]` part is misleading. We're not running our `entrypoint.sh`, but the [one coming from the base image](https://github.com/kubernetes-sigs/kwok/blob/main/images/cluster/entrypoint.sh).

With this change, are we still starting it?

Comment (yftacherzog) [reply kwok/Dockerfile:15]: (or still running something that does a similar job)

Comment (srivickynesh) [reply kwok/Dockerfile:15]: yes @yftacherzog .

Comment (ifireball) [reply kwok/Dockerfile:15]: no, we are overriding the one in the base image with our own during the image build. We have no need for the functionality in the base image entry point which has to do with setting up a cluster which we do during the build.

Comment (srivickynesh) [reply kwok/Dockerfile:15]: Only change why it couldn't reload inittab might be on line 15. https://github.com/srivickynesh/segment-bridge/pull/1/files

Comment (yftacherzog) [reply kwok/Dockerfile:15]: I don't think our `entrypoint.sh` is being copied to the image, so the existing functionality is not using those lines that this PR is putting in `kwok/inittab`.

Comment (ifireball) [reply kwok/Dockerfile:15]: @srivickynesh you are not running init at all in that patch, and you've added a lot of unnecessary stuff that probably causes issues such as command for starting TTYs.

Comment (srivickynesh) [reply kwok/Dockerfile:15]: those are all the defaults that comes within inittab

Comment (yftacherzog) [reply kwok/Dockerfile:15]: ok. I realize now that the line of copying the entrypoint was not a part of the code I was dealing with yesterday.

File: kwok/Dockerfile [hunk 2]
```diff
@@ -4,11 +4,12 @@ ARG CLUSTER_NAME=kwok
 ENV CLUSTER_NAME=${CLUSTER_NAME}
 
 COPY --chmod=755 user-signups.sh entrypoint.sh toolchain.dev.openshift.com_usersignups.yaml ./
```
Comment (srivickynesh) [inline kwok/Dockerfile:6]:
```suggestion
COPY --chmod=755 user-signups.sh toolchain.dev.openshift.com_usersignups.yaml ./
```

Comment (srivickynesh) [reply kwok/Dockerfile:6]: Also we need to remove kwok/entrypoint.sh as part of this change.

**Review by srivickynesh — CHANGES_REQUESTED** [review]

File: kwok/Dockerfile (hunk 1 above)
Comment (ifireball) [reply kwok/Dockerfile:15]: @srivickynesh those defaults are for booting whole servers, you are not doing that here...

File: kwok/Dockerfile (hunk 2 above)
Comment (ifireball) [reply kwok/Dockerfile]: done

**Review by srivickynesh — APPROVED** [review]
LGTM

**Pull request merged**